### DISCUSSION TITLE
Bulk edit user groups

### DISCUSSION THREAD

**Discussion by KatieMFritz** [craftcms/cms#8877] [Ideas]
### Description
I want to be able to assign multiple users at once to different user groups.

### Use case:
We are managing permissions and member segmentation with a "Paper Reviewers" user group. After the current set of papers is done, we want to clear out that group so we can reuse it. We might want to add those people to a "Paper Reviewers - Past" user group for future reference (like a tag). 

Currently, I can do this when I'm editing an _individual_ user:
![Screen Shot 2020-07-29 at 11 02 52 AM](https://user-images.githubusercontent.com/7085184/88817047-128b4d80-d18b-11ea-93f0-3710b0f7e932.png)

But if I select multiple users from the element index view, I only have the following options:
![Screen Shot 2020-07-29 at 10 57 03 AM](https://user-images.githubusercontent.com/7085184/88817212-3d75a180-d18b-11ea-9378-809814395559.png)

MVP: A "remove from group" option would be great.

The dream: some kind of bulk permissions editor.

**If there's a workaround for now, I'm all ears!** 👂👂👂👂👂👂

**Comment by brandonkelly** [comment]
We should be able to get this in as part of #893. Just added that to the 3.6 milestone. 🤞🏻

**Comment by sjcallender** [comment]
We love you @brandonkelly.

**Comment by KatieMFritz** [comment]
Looks like this is now scheduled for 4.0, not 3.6? 

Any tips for a workaround in the meantime? Do we just need to write a module for it?

**Comment by brandonkelly** [comment]
Yeah plans changed a bit. The same release that originally was going to be “3.6” is now “4.0”, which is under development.

There will be a 3.6 (released any day now) but it’s significantly smaller in scope – mostly just dependency updates focused on PHP 8 compatibility.

**Comment by ademers** [comment]
I built and manage a membership site and have created several logical groups to better organize all the users. They need to be periodically removed from certain groups. I'm currently doing this via a Console Controller in a Custom module that I run from the CLI. However, being to do this from the CP would bee a lot easier and would also allow others to do this instead of me.

![CleanShot 2021-04-08 at 15 07 32](https://user-images.githubusercontent.com/901095/114083900-90a50580-987d-11eb-9e04-eabb86d2b093.jpeg)

**Comment by whoisgina** [comment]
This feature would be a game changer for two of my clients! One is managing access to community resources across multiple programs with multiple user groups, and another will be managing complex subscriptions and memberships on a Commerce site. I'll look into resaving users via the CLI, but ideally admins can do this themselves in the CP as groups and programs change frequently.
